src/librustc/infer/mod.rs

```diff
@@ -410,15 +410,17 @@ pub enum NLLRegionVariableOrigin {
     /// from a `for<'a> T` binder). Meant to represent "any region".
     Placeholder(ty::PlaceholderRegion),
 
-    Existential,
+    Existential {
+        was_placeholder: bool
```

Comment:
I don't think this is quite the right name, per my prior comments. This existential didn't come from a placeholder, or at least not placeholder in the (rather narrow) sense that I aim to use the term. I'm trying to decide what name I would prefer, though. Perhaps something like `from_hr_binders`? I guess that's pretty obscure. Perhaps `from_forall` or `from_for_binder` or just `from_for`? It would then include a comment like:

```rust
Existential {
    /// If this is true, then this variable was created to represent a lifetime
    /// bound in a `for` binder. For example, it might have been created to
    /// represent the lifetime `'a` in a type like `for<'a> fn(&'a u32)`.
    /// Such variables are created when we are trying to figure out if there
    /// is any valid instantiation of `'a` that could fit into some scenario.
    ///
    /// This is used to inform error reporting: in the case that we are trying to
    /// determine whether there is any valid instantiation of a `'a` variable that meets
    /// some constraint C, we want to blame the "source" of that `for` type, 
    /// rather than blaming the source of the constraint C.
    from_forall: bool,
}
```

Reply:
The reason for me being persnickety here is that there is a subtle distinction between the region in a `for<'a>` binder and the placeholder that we (sometimes) instantiate it with, and I'd prefer not to blur those lines, since it can be hard enough to explain as is.